lib/api/objectCopy.js

```diff
@@ -140,28 +141,46 @@ function objectCopy(authInfo, request, sourceBucket,
     }
 
     return async.waterfall([
-        function checkSourceAuthorization(next) {
+        function checkDestAuth(next) {
+            return services.metadataValidateAuthorization(valPutParams,
```

Comment:
A comment Lauren made when I discussed this with her is that the other checks are all synchronous, and it's a bit odd to leave them hanging there in the waterfall when no further asynch call needs to be made.

The concern seems to be if there was a particular order to the checks for a reason, but there isn't much evidence for that reason. We don't have any functional tests that show the order matters at least. The error is similar for both source and destination auth failure (both result in AccessDenied).